### ISSUE TITLE
Not receiving messages between atsigns using at_onboarding_cli+at_auth

### ISSUE THREAD

**Issue by purnimavenkatasubbu** [atsign-foundation/at_talk#19]
### Describe the bug

Sent messages are not being displayed on the receiver's end

### Steps to reproduce

1. Clone the trunk branch of at_talk (change the dart version)
2. Then Add the following dependency overrides 
   dependency_overrides: 
  at_commons: ^3.0.56
  at_lookup: ^3.0.40
  at_client:
    git:
      url: https://github.com/atsign-foundation/at_client_sdk
      ref: trunk
      path: packages/at_client
  at_onboarding_cli:
    git:
      url: https://github.com/atsign-foundation/at_libraries.git
      path: packages/at_onboarding_cli
      ref: apkam_onboarding_changes

4. And then generate a new atsigns with register_cli(apkam_onbording_changes) branch
5.  Run the following commands for the 2 atsigns

```
 dart bin/at_talk.dart -k /home/shaikirfan/.atsign/keys/@yeastydotterel85_key.atKeys -a @yeastydotterel85 -t @charlie24

dart bin/at_talk.dart -k /home/shaikirfan/.atsign/keys/@charlie24_key.atKeys -a @charlie24 -t @yeastydotterel85
```

[atTalkLOgs.txt](https://github.com/atsign-foundation/at_talk/files/12893546/atTalkLOgs.txt)

[attalkreceiverlogs.txt](https://github.com/atsign-foundation/at_talk/files/12893565/attalkreceiverlogs.txt)
![Screenshot from 2023-10-13 18-11-29](https://github.com/atsign-foundation/at_talk/assets/43639593/218ce524-d24d-479c-bf4e-187b70fbd19f)

### Expected behavior

Messages sent and received should be displayed.

### Screenshots

![Screenshot from 2023-10-13 18-11-29](https://github.com/atsign-foundation/at_talk/assets/43639593/6c162783-facd-4a08-9189-98c7e95742c8)

https://github.com/atsign-foundation/at_talk/assets/43639593/27b776d5-f047-4c6d-8a67-4d2352702144

**Comment by purnimavenkatasubbu** [comment]
The issue got fixed by updating AtOnboardingPreference()..useAtChops = true
atsign-1

dart bin/at_talk.dart -k /home/shaikirfan/.atsign/keys/@ yeastydotterel85_key.atKeys -a @ yeastydotterel85 -t @ charlie24
Connecting ... Connected
@ charlie24: hello yeasty
@ yeastydotterel85: hi charlie
@ charlie24: how are you?
@ yeastydotterel85: i'm fine.. what about you?
@ charlie24: doing great
@ charlie24: 123
@ charlie24: 18793
@ charlie24: hellow
@ charlie24: are you fine
@ yeastydotterel85: yayy yes 
@ yeastydotterel85: cool
@ yeastydotterel85: hello
@ yeastydotterel85: 123
@ yeastydotterel85: 78172817
@ yeastydotterel85: hello world
@ yeastydotterel85: 
@ yeastydotterel85: yes yes

atsign-2
dart bin/at_talk.dart -k /home/shaikirfan/.atsign/keys/@ charlie24_key.atKeys -a @ charlie24 -t @ yeastydotterel85
Connecting ... Connected
@ charlie24: hello yeasty
@ yeastydotterel85: hi charlie
@ charlie24: how are you?
@ yeastydotterel85: i'm fine.. what about you?
@ charlie24: doing great
@ charlie24: 123
@ charlie24: 18793
@ charlie24: hellow
@ charlie24: are you fine
@ yeastydotterel85: yayy yes
@ yeastydotterel85: cool

**Comment by murali-shris** [comment]
@gkc at_auth package requires at_chops for pkam signing.
monitor.dart has reading private key from preference as a fallback option.We no longer set privatekey in preference and if useAtChops=false, monitor is failing.

Fix for this issue has to be in at_talk.dart..is this fine?

**Comment by gkc** [comment]
@murali-shris Shouldn't we instead make useAtChops true by default in at_client? `useAtChops` was only ever intended as a temporary thing so we could gracefully and safely migrate from the pre-atChops approach

**Comment by murali-shris** [comment]
> @murali-shris Shouldn't we instead make useAtChops true by default in at_client? `useAtChops` was only ever intended as a temporary thing so we could gracefully and safely migrate from the pre-atChops approach

yes Gary. We can. We should test there are no legacy code in packages that sets private key in preferences and not setting atChops.

**Comment by gkc** [comment]
@murali-shris OK please create a P1 ticket for this work asap

**Comment by murali-shris** [comment]
> @murali-shris OK please create a P1 ticket for this work asap

https://github.com/atsign-foundation/at_client_sdk/issues/1141
I have started working on this task

**Comment by murali-shris** [comment]
issue fixed and at_client 3.0.66 published
